packages/flutter/test/material/toggle_buttons_test.dart

```diff
@@ -1398,6 +1398,155 @@ void main() {
+        Material(
+          child: boilerplate(
+            child: ToggleButtons(
+              direction: Axis.vertical,
+              verticalDirection: VerticalDirection.down,
+              isSelected: const <bool>[false, true, false],
+              onPressed: (int index) {},
+              children: const <Widget>[
+                Text('First child'),
+                Text('Second child'),
+                Text('Third child'),
+              ],
+            ),
+          ),
+        ),
+      );
+
+      // The children should be laid out along vertical and the first child at top.
+      expect(tester.getCenter(find.text('First child')), const Offset(400.0, 251.0));
+      expect(tester.getCenter(find.text('Second child')), const Offset(400.0, 300.0));
+      expect(tester.getCenter(find.text('Third child')), const Offset(400.0, 349.0));
+
+      final List<RenderObject> toggleButtonRenderObject = tester.allRenderObjects.where((RenderObject object) {
+        return object.runtimeType.toString() == '_SelectToggleButtonRenderObject';
+      }).toSet().toList();
+
+      // The first button paints the left, top and right sides with a path.
+      expect(
+        toggleButtonRenderObject[0],
+        paints
+        // left side, top and right - enabled.
+          ..path(
+            style: PaintingStyle.stroke,
+            color: theme.colorScheme.onSurface.withOpacity(0.12),
+            strokeWidth: _defaultBorderWidth,
+          ),
+      );
+
+      // The middle buttons paint a top side path first, followed by a
+      // left and right side path.
+      expect(
+        toggleButtonRenderObject[1],
+        paints
+        // top side - selected.
+          ..path(
+            style: PaintingStyle.stroke,
+            color: theme.colorScheme.onSurface.withOpacity(0.12),
+            strokeWidth: _defaultBorderWidth,
+          )
+        // left and right - selected.
+          ..path(
+            style: PaintingStyle.stroke,
+            color: theme.colorScheme.onSurface.withOpacity(0.12),
+            strokeWidth: _defaultBorderWidth,
+          ),
+      );
+
+      // The last button paints a top side path first, followed by
+      // a left, bottom and right side path
+      expect(
+        toggleButtonRenderObject[2],
+        paints
+        // top side - selected, since previous button is selected.
+          ..path(
+            style: PaintingStyle.stroke,
+            color: theme.colorScheme.onSurface.withOpacity(0.12),
+            strokeWidth: _defaultBorderWidth,
+          )
+        // left side, bottom and right - enabled.
+          ..path(
+            style: PaintingStyle.stroke,
+            color: theme.colorScheme.onSurface.withOpacity(0.12),
+            strokeWidth: _defaultBorderWidth,
+          ),
+      );
+    },
+  );
+
+  testWidgets(
+    'VerticalDirection test when direction is vertical.',
+        (WidgetTester tester) async {
+      await tester.pumpWidget(
```

Comment:
Would this indentation look better for the tests?
```suggestion
  testWidgets(
    'VerticalDirection test when direction is vertical.',
    (WidgetTester tester) async {
      await tester.pumpWidget(
```